spec/lib/puppet/test_ca.rb

```diff
@@ -123,10 +123,23 @@ def generate(name, opts)
 
     private
 
-    def build_cert(name, issuer)
-      key = OpenSSL::PKey::RSA.new(1024)
+    def build_cert(name, issuer, opts = {})
+      key = if opts[:key_type] == :ec
+              key = OpenSSL::PKey::EC.generate('prime256v1')
```

Comment:
Added a test to `state_machine_spec` that it can generate the default `prime256p1`, `secp384r1`, and that an exception is raised if an unknown curve is specified.